### PR TITLE
 self update after updating a specific toolchain 

### PR DESCRIPTION
**Pull request by nrc**
r? @alexcrichton 

This will make sure people who do `rustup update stable` still get a new rustup

### CODE REVIEW

**Comment by alexcrichton** [comment]
r=me with green CI

File: src/rustup-cli/rustup_mode.rs [hunk 1]
```diff
@@ -132,6 +132,19 @@ pub fn cli() -> App<'static, 'static> {
                         .help(TOOLCHAIN_ARG_HELP)
                         .required(true)
                         .multiple(true),
+                )
+                .arg(
+                    Arg::with_name("no-self-update")
+                        .help("Don't perform self update when running the `rustup` command")
```
Comment (kinnison) [inline src/rustup-cli/rustup_mode.rs:138]: should this be:

```rust
.help("Don't perform self update when running the `rustup install` command")
```

perhaps?

File: src/rustup-cli/rustup_mode.rs [hunk 2]
```diff
@@ -196,6 +209,13 @@ pub fn cli() -> App<'static, 'static> {
                                 .help(TOOLCHAIN_ARG_HELP)
                                 .required(true)
                                 .multiple(true),
+                        )
+                        .arg(
+                            Arg::with_name("no-self-update")
+                                .help("Don't perform self update when running the `rustup` command")
```
Comment (kinnison) [inline src/rustup-cli/rustup_mode.rs:215]: Ditto here but with 
```rust
.help("Don't perform self update when running the `rustup toolchain install` command")
```

perhaps?

**Review by kinnison — APPROVED** [review]
I'm pretty OK with this in general, though the amount of tests needing `--no-self-update` I wonder if it might be better to also have a `RUSTUP_SUPPRESS_SELF_UPDATE` type environment variable which we could set by default in most of the `expect_ok()` calls.

Nothing blocking a merge IMO though.

**Pull request merged**